pkg/apis/v1/nodepool_validation_cel_test.go

```diff
@@ -695,4 +696,182 @@ var _ = Describe("CEL/Validation", func() {
 			Expect(env.Client.Create(ctx, nodePool)).ToNot(Succeed())
 		})
 	})
+	Context("Replicas", func() {
+		Context("Valid Replicas Values", func() {
+			It("should succeed when replicas is set to a positive value", func() {
+				nodePool.Spec.Replicas = lo.ToPtr(int64(5))
+				Expect(env.Client.Create(ctx, nodePool)).To(Succeed())
+				Expect(nodePool.RuntimeValidate(ctx)).To(Succeed())
+			})
+
+			It("should succeed when replicas is set to zero", func() {
+				nodePool.Spec.Replicas = lo.ToPtr(int64(0))
+				Expect(env.Client.Create(ctx, nodePool)).To(Succeed())
+				Expect(nodePool.RuntimeValidate(ctx)).To(Succeed())
+			})
+
+			It("should succeed when replicas is set to a large value", func() {
+				nodePool.Spec.Replicas = lo.ToPtr(int64(1000))
+				Expect(env.Client.Create(ctx, nodePool)).To(Succeed())
+				Expect(nodePool.RuntimeValidate(ctx)).To(Succeed())
+			})
+		})
+
+		Context("Invalid Replicas Values", func() {
+			It("should fail when replicas is set to a negative value", func() {
+				nodePool.Spec.Replicas = lo.ToPtr(int64(-100))
+				Expect(env.Client.Create(ctx, nodePool)).ToNot(Succeed())
+			})
+		})
+
+		DescribeTable("In-compatible fields for static nodepools",
```

Comment:
nit: DescribeTable should still follow the "should ..." format
```suggestion
		DescribeTable("should fail for incompatible fields",
```